### DISCUSSION TITLE
What is the point of ConnectionStrings section in appsettings.json?

### DISCUSSION THREAD

**Discussion by MhAllan** [dotnet/aspnetcore#29600] [Q&A] [answered]
What is the point of ConnectionStrings section in appsettings.json?

**Comment by gfoidl** [comment, marked as answer]
To specify connection strings (to databases).

Connection strings are used quite often, so there's a handy extension-method [GetConnectionString](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.configuration.configurationextensions.getconnectionstring?view=dotnet-plat-ext-5.0) to simplify retrieval of it's value. 
See e.g. [EF Core Docs for Connection Strings](https://docs.microsoft.com/en-us/ef/core/miscellaneous/connection-strings) and [Connection string prefixes](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-5.0#connection-string-prefixes) in the ASP.NET Core Docs.

**Reply by MhAllan** [reply]
I don't think it worth to be recognized section, let people define the way they want, because when I want to set a connection string I maybe want to set some other related configurations like timeout and database name ... etc. so I will be putting them together, this section is weird for me unless it has some benefits in devops that I can't see

**Reply by davidfowl** [reply]
It's a convention, you don't need to follow it.
